textalyze.js

```diff
@@ -39,6 +39,31 @@ function itemCounts(array) {
   }, new Map());
 }
 
+/**
+ * Given a count of chars in a text, gets its map of chars and occurences, and calculates a map of chars and frequencies based
+ * on the passed count of chars.
+ * @param {Integer} totalCount - The total amount of chars in a text.
+ * @param {Map} itemCounts - A map containing the occurrences of each char in the text.
+ * @returns {Map} itemFrequencies - A map containing the frequencies of each char based on the total count.
+ */
+function itemFrequencies(totalCount, itemCounts) {
+  if (totalCount <= 0) {
+    throw new Error('The items frequencies can\'t be computed without a total count of chars.');
+  }
+
+  if (typeof itemCounts != 'object' || !(itemCounts instanceof Map)) {
+    throw new Error('The item frequencies must be a valid map.');
```

Comment:
Here you'd want [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError).